nav2_smoother/test/test_simple_smoother.cpp

```diff
@@ -248,7 +244,7 @@ TEST(SmootherTest, test_simple_smoother)
   straight_irregular_path.poses[6].pose.position.x = 0.5;
   straight_irregular_path.poses[6].pose.position.y = 0.5;
   straight_irregular_path.poses[6].pose.orientation = tf2::toMsg(quat2);
-  EXPECT_TRUE(smoother->smooth(straight_irregular_path, max_time));
+  EXPECT_THROW(smoother->smooth(straight_irregular_path, max_time), nav2_core::FailedToSmoothPath);
```

Comment:
Adding the failed to smooth exception exposes that this test currently doesn't do anything. 

The path segments are less than 9 thus nothing gets smoothed. 
https://github.com/ros-planning/navigation2/blob/main/nav2_smoother/src/simple_smoother.cpp#L74

The reason it returns true is because success is initialized to true.

Reply:
Got it - well I think this update makes sense, we can keep it :+1: